### PR TITLE
ESLinting

### PR DESCRIPTION
**Pull request by mtimofiiv**
Added some linting rules + linted all the files so far.

Feel free to make suggestions of rules I missed or ones you want added!

Also added on @adriaanvanrossum's suggestion to have the linter run before the test suite.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    "no-plusplus": ["off"],
+    "arrow-parens": ["off"],
+    "no-restricted-syntax": ["off"],
+    "no-console": ["off"],
+    "no-continue": ["off"],
+    "import/no-unresolved": [2, {}],
+    "import/prefer-default-export": ["off"],
+    "class-methods-use-this": ["off"]
+  }
+}
```
Comment (adriaandotcom) [inline .eslintrc.js]: Line endings? This could help: `eol-last: ["error", "always"]`

https://eslint.org/docs/rules/eol-last

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  },
+  "rules": {
+    "global-require": ["off"],
+    "comma-dangle": ["error", "never"],
+    "no-plusplus": ["off"],
+    "arrow-parens": ["off"],
+    "no-restricted-syntax": ["off"],
+    "no-console": ["off"],
+    "no-continue": ["off"],
+    "import/no-unresolved": [2, {}],
```
Comment (adriaandotcom) [inline .eslintrc.js]: Why is this? I found this super useful

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
+module.exports = {
```
Comment (adriaandotcom) [inline .eslintrc.js:1]: In the frontend we use this

```js
module.exports = {
  root: true,
  parserOptions: {
    ecmaVersion: 2017,
    sourceType: 'module'
  },
  extends: 'eslint:recommended',
  env: {
    browser: true,
    es6: true
  },
  rules: {
    "arrow-parens": ["error", "as-needed"],
    "comma-dangle": ["error", "never"],
    "no-console": ["warn", { allow: ["error"] }]
  }
};
```

**Review by adriaandotcom — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (mtimofiiv) [reply .eslintrc.js]: Added!

File: .eslintrc.js (hunk 2 above)
Comment (mtimofiiv) [reply .eslintrc.js]: This does not turn off this rule. It merely tells the plugin to use the default settings.

File: .eslintrc.js (hunk 3 above)
Comment (mtimofiiv) [reply .eslintrc.js:1]: I'm ok with these - I added the `env`s (+ a `node` one as well), as well as `no-console` and changed `arrow-parens`.

**Review by andreamarchetti — APPROVED** [review]

**Comment by mtimofiiv** [comment]
waiting on @adriaanvanrossum to re-review.....

**Review by adriaandotcom — APPROVED** [review]

**Pull request merged**